File: rb/features.rb
# ForeignExchangeRates SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module ForeignExchangeRatesFeatures
  def self.make_feature(name)
    case name
    when "base"
      ForeignExchangeRatesBaseFeature.new
    when "test"
      ForeignExchangeRatesTestFeature.new
    else
      ForeignExchangeRatesBaseFeature.new
    end
  end
end

File: py/features.py
# ForeignExchangeRates SDK feature factory

from feature.base_feature import ForeignExchangeRatesBaseFeature
from feature.test_feature import ForeignExchangeRatesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ForeignExchangeRatesBaseFeature(),
        "test": lambda: ForeignExchangeRatesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
